imgui-binding/src/main/java/imgui/extension/imguizmo/ImGuizmo.java

```diff
@@ -0,0 +1,240 @@
+        nEnabled(buffer);
+    }
+
+    private native static void nIsUsing(boolean[] buffer); /*
+        buffer[0] = ImGuizmo::IsUsing();
+    */
+
+    /**
+     * Natively checks to see if we're using the guizmo
+     */
+    public static boolean isUsing() {
+        nIsUsing(buffer);
+        return buffer[0];
+    }
+
+    private native static void nIsOver(boolean[] buffer); /*
+        buffer[0] = ImGuizmo::IsOver();
+    */
+
+    /**
+     * Natively checks to see if we're over the guizmo
+     */
+    public static boolean isOver() {
+        nIsOver(buffer);
+        return buffer[0];
+    }
+
+    private native static void nSetDrawList(long pointer); /*
+        ImGuizmo::SetDrawlist((ImDrawList*)pointer);
+    */
+
+    /**
+     * This will set the draw list of the given guizmo
+     */
+    public static void setDrawList(ImDrawList drawList) {
+        nSetDrawList(drawList.ptr);
+    }
+
+
+    /**
+     * This will set the default window drawlist value
+     */
+    public static void setDrawList() {
+        setDrawList(ImGui.getWindowDrawList());
+    }
+
+    private native static void nBeginFrame(); /*
+        ImGuizmo::BeginFrame();
+    */
+
+    /**
+     * Starts the next frame for the guizmo
+     */
+    public static void beginFrame() {
+        nBeginFrame();
+    }
+
+    private native static void nDecomposeMatrixToComponents(float[] matrix, float[] translation, float[] rotation, float[] scale); /*
+        ImGuizmo::DecomposeMatrixToComponents(matrix, translation, rotation, scale);
+    */
+
+    /**
+     * This will allow for manual editing of the guizmos
+     */
+    public static void decomposeMatrixToComponents(float[] matrix, float[] translation, float[] rotation, float[] scale) {
+
+        nDecomposeMatrixToComponents(matrixBuffer, positionBuffer, rotationBuffer, scaleBuffer);
+
+
+    }
+
+    private native static void nRecomposeMatrixFromComponents(float[] translation, float[] rotation, float[] scale, float[] matrix); /*
+        ImGuizmo::RecomposeMatrixFromComponents(translation, rotation, scale, matrix);
+    */
+
+
+    /**
+     * This will allow for manual editing of the guizmos
+     */
+    public static void recomposeMatrixFromComponents(float[] matrix, float[] translation, float[] rotation, float[] scale) {
+
+
+        nRecomposeMatrixFromComponents(positionBuffer, rotationBuffer, scaleBuffer, matrixBuffer);
+
+
+    }
+
+    public native static void nSetRect(float x, float y, float width, float height); /*
+        ImGuizmo::SetRect(x, y, width, height);
+    */
+
+    /**
+     * This will set the rect position
+     */
+    public static void setRect(float x, float y, float width, float height) {
+        nSetRect(x, y, width, height);
+    }
+
+
+    /**
+     * This will set the rect position to the default current rect position
+     */
+    public static void setRect() {
+        ImVec2 pos = ImGui.getWindowPos();
+        ImVec2 size = ImGui.getWindowSize();
```

Comment:
Both of those methods create new objects. In binding I avoid unnecessary allocations as much as possible. You should do something like that https://github.com/SpaiR/imgui-java/blob/0ed891f391f70e1ba4edf863a6f8d22ccca23c98/imgui-binding/src/main/java/imgui/internal/ImGuiDockNode.java#L651-L660